### DISCUSSION TITLE
lotem/rime-octagram-data-s1 如何使用?

### DISCUSSION THREAD

**Discussion by bennyyip** [rime/home#1306] [Q&A]
项目提供了[zh-hant-t-s1-bgc.gram](https://github.com/lotem/rime-octagram-data-s1/blob/master/zh-hant-t-s1-bgc.gram) 和 [zh-hant-t-s1-tgc.gram](https://github.com/lotem/rime-octagram-data-s1/blob/master/zh-hant-t-s1-tgc.gram) 两个模型文件，`tgc`和`bgc`分别是什么意思？
由于不了解S1模型的含义，`translator/{min,max}_homophones` 和 `collocation_{min,max}_length` 的参数我不知道要填多少。可否提供一份类似 https://github.com/lotem/rime-octagram-data/blob/master/grammar.yaml 的配置文件？

**Comment by lotem** [comment]
不是項目提供的。是個人發佈的，沒有經過質量檢驗。

`max_homophones` 表示每組同音字/詞，取多少個進行組句。一般不需要通過配置調整，可以用程序裏的默認值。
值越大則較生僻字詞的覆蓋率更高，運算量也隨之更大。同音字中排位靠後的字詞通常無法成句，成功率低，因此設置一個很大的值並不經濟。

`collocation_{min,max}_length` 設置比配詞組的漢字數範圍。最小值通常可以設成 2 即兩個單字的組合，最大值根據語言模型能提供的最大詞組長度決定。設得過大，用長字串查數據庫不會命中，白白浪費效能。

`bgc` 大概意思是以字爲單位的二元組，所以最大值設爲 2 就夠了。`tgc` 則設爲 3。
